### PR TITLE
virtual-hub: Component to connect multiple 802.15.4 qemu nodes

### PR DESCRIPTION
**Pull request by pedropaulomartucci**
Virtual-hub is a tool which connects multiple 802.15.4 qemu
nodes to simulate physical networks with different topologies.

### CODE REVIEW

**Comment by pedropaulomartucci** [comment]
This tool was developed while validating RPL global-repair scenarios on:
https://github.com/zephyrproject-rtos/zephyr/pull/5202

It was suggested me to submit this tool to net-tools repository.

File: virtual-hub/src/graphs.c [hunk 1]
```diff
@@ -0,0 +1,90 @@
+#include <stdio.h>
```
Comment (jukkar) [inline virtual-hub/src/graphs.c]: Could you add license / copyright information in relavant files, at least .c and .h files need this info.

File: virtual-hub/src/hub.c [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+{
+	for (int i = 0; i < fdSize; i++) {
+		if (send[pos][i] == TRUE) {
+			write(fd_in[i], buf, 1);
+		}
+	}
+}
+
+static gboolean fifo_handler(GIOChannel *channel,
+							GIOCondition cond,
```
Comment (jukkar) [inline virtual-hub/src/hub.c]: Something wrong with the indentation here.

File: virtual-hub/src/hub.c [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+			g_strconcat(str_num, PIPE_OUT, NULL), NULL);
+
+		fifoTrain[i] = setup_fifofd(pipe_out[i], i);
+
+		if (fifoTrain[i] > 0) {
+			open_fifo++;
+		} else {
+			printf("Failed to open the fifo node %d\n", (i + 1));
+			printf(
+				"******** Probably inserted wrong number of nodes or this node"
```
Comment (jukkar) [inline virtual-hub/src/hub.c]: Could you start the string in the same line as printf()?

File: virtual-hub/src/hub.c [hunk 4]
```diff
@@ -0,0 +1,385 @@
...
+		}
+	}
+
+	if (open_fifo == fdSize) {
+		printf("-------------------------------\n");
+		printf("All fifo opened with sucess!!\n");
+		printf("-------------------------------\n");
+	}
+
+	int currentIndex;
```
Comment (jukkar) [inline virtual-hub/src/hub.c]: Please introduce the variables at the start of the function.

File: virtual-hub/src/hub.c [hunk 5]
```diff
@@ -0,0 +1,385 @@
...
+			G_IO_IN | G_IO_HUP | G_IO_ERR | G_IO_NVAL,
+			fifo_handler, GINT_TO_POINTER(pos));
+
+	g_io_channel_unref(channel);
+	return source;
+}
+
+int main(int argc, char *argv[])
+{
+	srand((unsigned int) time(NULL));
```
Comment (jukkar) [inline virtual-hub/src/hub.c]: Put this after variable definitions.

**Comment by pedropaulomartucci** [comment]
@jukkar : Updated your requests.

**Comment by jukkar** [comment]
Looks good, could you squash your fixes into the relevant commit. We typically do not have "fix" commits inside one PR.

**Comment by pedropaulomartucci** [comment]
Done.

**Pull request merged**